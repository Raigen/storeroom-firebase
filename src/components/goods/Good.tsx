import React from 'react'
import Typography from '@material-ui/core/Typography'

export type GoodType = {
  _id: string
  name: string
  unit: string
}

type GoodListEntryProps = {
  good: GoodType
}

export const GoodListEntry: React.FC<GoodListEntryProps> = ({ good }) => {
  return (
    <Typography>
      {good.name} - {good.unit}
    </Typography>
  )
}
