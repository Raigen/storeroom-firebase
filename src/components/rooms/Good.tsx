import React from 'react'
import { useDocumentData } from 'react-firebase-hooks/firestore'
import firebase from 'firebase'
import { RoomGoodType } from './GoodsList'

type GoodType = {
  _id: string
  name: string
  unit: string
}

type GoodListEntryProps = { goodData: RoomGoodType }

export const GoodListEntry: React.FC<GoodListEntryProps> = ({ goodData }) => {
  const [good, loading, error] = useDocumentData<GoodType>(firebase.firestore().doc(`/goods/${goodData.ref.id}`), {
    idField: '_id'
  })
  if (error) return <li>{error.message}</li>
  if (loading || !good) return null
  return (
    <>
      {good.name} - {goodData.amount}
      {good.unit}
    </>
  )
}
