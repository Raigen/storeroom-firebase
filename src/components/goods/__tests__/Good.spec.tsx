import { fireEvent, render } from '@testing-library/react'

import { GoodListEntry } from '../Good'
import React from 'react'

it('should render a good with name and unit', function() {
  const deleteEntry = jest.fn().mockName('onDelete')
  const { container } = render(
    <GoodListEntry good={{ _id: '1', name: 'Test good', unit: 'kg' }} onDelete={deleteEntry} />
  )
  expect(container).toMatchInlineSnapshot(`
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Test good
         - 
        kg
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          title="Löschen"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </p>
    </div>
  `)
})

it('should delete a good', async function() {
  const deleteEntry = jest.fn().mockName('onDelete')
  const { container, getByTitle } = render(
    <GoodListEntry good={{ _id: '1', name: 'Test good', unit: 'kg' }} onDelete={deleteEntry} />
  )
  fireEvent.click(getByTitle('Löschen'))
  expect(deleteEntry).toHaveBeenCalledWith('1')
})
