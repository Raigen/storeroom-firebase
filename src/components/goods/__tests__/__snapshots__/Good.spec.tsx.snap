// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a good with name, unit and delete button 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Test good
     - 
    kg
    <button
      class="MuiButtonBase-root MuiIconButton-root"
      tabindex="0"
      title="LÃ¶schen"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </p>
</div>
`;

exports[`should render a good without delete button 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Test good
     - 
    kg
  </p>
</div>
`;
